### PR TITLE
Webhost time limit

### PR DESCRIPTION
**Pull request by Berserker66**
Please format your title with what portion of the project this pull request is
targeting and what it's changing.

ex. "MyGame4: implement new game" or "Docs: add new guide for customizing MyGame3"

## What is this fixing or adding?
Webhost generation time limit

## How was this tested?
with 1 second to trigger it, and with a long time to see it not trigger

## If this makes graphical changes, please attach screenshots.
image

### CODE REVIEW

File: worlds/factorio/__init__.py [hunk 1]
```diff
@@ -157,6 +157,8 @@ def set_rules(self):
                 satellite_recipe = self.custom_recipes["satellite"] if "satellite" in self.custom_recipes \
                     else next(iter(all_product_sources.get("satellite")))
             victory_tech_names = get_rocket_requirements(silo_recipe, part_recipe, satellite_recipe)
+            if self.multiworld.silo[self.player] != Silo.option_spawn:
+                victory_tech_names.add("rocket-silo")
             world.get_location("Rocket Launch", player).access_rule = lambda state: all(state.has(technology, player)
```
Comment (ThePhar) [inline worlds/factorio/__init__.py:162]: Sneaking in Factorio fixes? 👀

**Comment by beauxq** [comment]
I think a better way to do this might be a certain amount of time per world.

example: 60 seconds per world
 - a 2-player generation fails after 120 seconds
 - a 15-player generation fails after 900 seconds

So a small generation will fail faster and not waste time, but we can still allow for larger generations.

**Comment by Berserker66** [comment]
> I think a better way to do this might be a certain amount of time per world.
> 
> example: 60 seconds per world
> 
> * a 2-player generation fails after 120 seconds
> * a 15-player generation fails after 900 seconds
> 
> So a small generation will fail faster and not waste time, but we can still allow for larger generations.

The intention is to clear out generations that might be stuck, which is why the default is 10 minutes. The complexity is meant to be controlled with the world count limit. I don't really like overlaying the two.

**Review by ThePhar — APPROVED** [review]
Generation time isn't linear with player counts anyway, so this should be fine for infinite loop situations.

**Pull request merged**